device/pensando/arm64-elba-asic-r0/plugins/ssd_util.py

```diff
@@ -0,0 +1,100 @@
+#
+# ssd_generic.py
+#
+# Generic implementation of the SSD health API
+# SSD models supported:
+#  - InnoDisk
+#  - StorFly
+#  - Virtium
+
+try:
+    import re
+    import subprocess
+    from sonic_platform_base.sonic_ssd.ssd_base import SsdBase
+except ImportError as e:
+    raise ImportError (str(e) + "- required module not found")
+
+NOT_AVAILABLE = "N/A"
+MMC_DATA_PATH = "/sys/class/mmc_host/mmc0/mmc0:0001/{}"
+
+class SsdUtil(SsdBase):
+    """
+    Generic implementation of the SSD health API
+    """
+    model = NOT_AVAILABLE
+    serial = NOT_AVAILABLE
+    firmware = NOT_AVAILABLE
+    temperature = NOT_AVAILABLE
+    health = NOT_AVAILABLE
+    ssd_info = NOT_AVAILABLE
+    vendor_ssd_info = NOT_AVAILABLE
+
+    def __init__(self, diskdev):
+
+        self.dev = diskdev
+        try:
+            self.model = ("emmc {}".format(open(MMC_DATA_PATH.format("name")).read())).replace("\n", "")
+            self.serial = open(MMC_DATA_PATH.format("serial")).read().replace("\n", "")
+            self.firmware = open(MMC_DATA_PATH.format("fwrev")).read().replace("\n", "")
+            value =  open(MMC_DATA_PATH.format("life_time")).read().replace("\n", "")
+            [lifetime_a, lifetime_b] = [int(val, 16) for val in value.split()]
+            lifetime = lifetime_a if lifetime_a >= lifetime_b else lifetime_b
+            self.health = float(100 - (lifetime*10))
```

Comment:
@ashwin-h for EMMC please use https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sonic_ssd/ssd_emmc.py

Reply:
> @ashwin-h for EMMC please use https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sonic_ssd/ssd_emmc.py

@shanshri - did this (above) get used per @prgeor 's request?

Reply:
We will address these things in the when will enable pmon in upstream

Reply:
@shanshri noted